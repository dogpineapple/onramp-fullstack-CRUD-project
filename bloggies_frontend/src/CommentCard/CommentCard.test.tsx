import React, { Component } from 'react';
import Enzyme, { mount, ReactWrapper, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CommendCard from './index';
import { MOCK_COMMENT } from '../jest.mock';

Enzyme.configure({ adapter: new Adapter() });

const mockHandlePostReply = jest.fn((post) => {
  return undefined;
})

describe('CommentCard', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(shallow<Component>(
      <CommendCard comment={MOCK_COMMENT} handlePostReply={mockHandlePostReply} />
    ).get(0));
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display comment', () => {
    const isCommentBody = wrapper.containsMatchingElement(
      <p className="sc-bdfBwQ dhWHrV card-text">
        test comment
      </p>
    );
    expect(isCommentBody).toBeTruthy();
  });
});