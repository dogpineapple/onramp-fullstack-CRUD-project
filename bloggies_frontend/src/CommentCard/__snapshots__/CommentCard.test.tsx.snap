// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard should match snapshot 1`] = `
<div
  className="CommentCard text-left"
>
  <Card
    body={false}
  >
    <div
      className="card"
    >
      <CardBody>
        <div
          className="card-body"
        >
          <CardText>
            <p
              className="card-text"
            >
              test comment
            </p>
          </CardText>
          <div
            className="text-muted d-flex justify-content-between"
          >
            <span
              className="d-flex flex-row align-items-center"
            >
              <span
                className="App-author ml-2 mr-1"
              >
                test user
              </span>
              commented 
              3 months ago
            </span>
            <Button
              active={false}
              disabled={false}
              onClick={[Function]}
              variant="primary"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Reply
              </button>
            </Button>
          </div>
        </div>
      </CardBody>
    </div>
  </Card>
</div>
`;
